NBitcoin/IpExtensions.cs

```diff
@@ -138,6 +138,36 @@ public static bool IsRFC4843(this IPAddress address)
 			return (bytes[15 - 15] == 0x20 && bytes[15 - 14] == 0x01 && bytes[15 - 13] == 0x00 && (bytes[15 - 12] & 0xF0) == 0x10);
 		}
 
+		public static byte[] GetGroup(this EndPoint endpoint)
+		{
+			if (endpoint is IPEndPoint ip)
+			{
+				return ip.Address.GetGroup();
+			}
+			else if (endpoint is DnsEndPoint dns)
+			{
+				var vchRet = new List<byte>();
+				if (dns.IsTor() || dns.IsI2P())

```

Comment:
Nit. This can be simplified. Remove this condition and check vchRet.Count. Do the nBits thing if Count != 0 otherwise exception.

Reply:
Sorry, I don't understand what you say. Could you explain me with (pseudo)code?

Reply:
```C#
			if (endpoint is DnsEndPoint dns)
			{
				var vchRet = new List<byte>();
				if (dns.IsTor())
				{
					vchRet.Add((byte)1);
				}
				else if (dns.IsI2P())
				{
					vchRet.Add((byte)2);
				}

				if (vchRet.Any())
				{
					var nBits = 4;
					var hostBytes = Encoding.UTF8.GetBytes(dns.Host);
					var b = hostBytes[0] | ((1 << (8 - nBits)) - 1);
					vchRet.Add((byte)b);
					return vchRet.ToArray();
				}
			}
```

Reply:
I think that even if it is not tor or i2p it still has to return a group.

Reply:
I see, ACK.